File: src/main/java/pl/training/workshop/m5/s14_reuse/start/LoyaltyReport.java
package pl.training.workshop.m5.s14_reuse.start;

/** Start: raport potrzebuje tylko właściciela i punktów - a przyjmuje całe LoyaltyAccount. */
public final class LoyaltyReport {
    public String line(LoyaltyAccount account) {
        return account.owner() + ": " + account.points() + " pkt";
    }
}

File: typescript/src/workshop/m5/s14_reuse/start/LoyaltyReport.ts
import type { LoyaltyAccount } from './LoyaltyAccount.js';

/** Start: raport potrzebuje tylko właściciela i punktów - a przyjmuje całe LoyaltyAccount. */
export class LoyaltyReport {
  line(account: LoyaltyAccount): string {
    return account.owner() + ': ' + account.points() + ' pkt';
  }
}
